tests/components/egps/test_init.py

```diff
@@ -0,0 +1,75 @@
+"""Tests for setting up egps integration."""
+
+from collections.abc import Generator
+from unittest.mock import MagicMock, patch
+
+from pyegps.exceptions import UsbError
+import pytest
+
+from homeassistant.components.egps.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(name="mock_get_device")
+def patch_get_device(pyegps_device_mock: MagicMock) -> Generator[MagicMock, None, None]:
+    """Fixture to patch the `get_device` api method."""
+    with patch(
+        "homeassistant.components.egps.get_device", return_value=pyegps_device_mock
+    ) as mock:
+        yield mock
```

Comment:
```suggestion
@pytest.fixture(name="mock_get_device")
def patch_get_device(pyegps_device_mock: MagicMock) -> Generator[MagicMock, None, None]:
    """Fixture to patch the `get_device` api method."""
    with patch(
        "homeassistant.components.egps.get_device"
    ) as mock,
    patch(
        "homeassistant.components.egps.get_device", new=mock
    ) :
        mock.return_value =pyegps_device_mock
        yield mock
```
not 100% sure, but I believe this should work and would allow us to move this to `conftest` to only define it once

Reply:
Sorry, I don't understand what you are doing here. Should the second patch be against "homeassistant.components.egps.config_flow.get_device" ? I could also import get_device from homeassistant.components.egps inside config_flow.py.  

In my opinion code duplication is sometimes acceptable if it helps readability of the code.

Reply:
yes, I meant the config flow in the second patch, sorry for that.

> In my opinion code duplication is sometimes acceptable if it helps readability of the code.

I agree with that, but in this case I don't see the benefit of defining the fixture multiple times

Reply:
ok, moved all fixtures to conftest.py